### PR TITLE
Add reader conditional support

### PR DESCRIPTION
**Pull request by JJ-Atkinson**
And a couple other changes.

### CODE REVIEW

File: src/compliment/context.clj [hunk 1]
```diff
@@ -43,6 +43,11 @@
                (str/replace "\\}" "(char 125)")
                (str/replace "{" "(compliment-hashmap ")
                (str/replace "}" ")")
+
+               ;; Basic enhance reader for forms with reader conditionals. Include the meta
+               ;; for sources that are sensitive to this.
+               (str/replace "#?" "^:reader-conditional? ")
+
```
Comment (JJ-Atkinson) [inline src/compliment/context.clj:50]: `{:read-cond :allow}` didn't have the behavior I wanted... It inevitably read the `:clj` branch of the conditional. I write a bunch of CLJC with specific :cljs branches, but the :cljs branches I'd like to still use completions for haha.

File: src/compliment/core.clj [hunk 2]
```diff
@@ -66,9 +66,10 @@
                                    (all-sources sources)
                                    (all-sources)))
              candidates (mapcat (fn [f] (f prefix nspc ctx)) candidate-fns)
-             sorted-cands (if (= sort-order :by-name)
-                            (sort-by :candidate candidates)
-                            (sort-by :candidate by-length-comparator candidates))
+             sorted-cands (case sort-order 
+                            :by-name (sort-by :candidate candidates)
+                            :by-length (sort-by :candidate by-length-comparator candidates)
+                            candidates)
```
Comment (JJ-Atkinson) [inline src/compliment/core.clj:72]: If this is unlikely to break clients, I'd love to make this change. Currently I have a keyword completer powered by pathom indexes that does its own sorting as part of the completion... I'd like to have an option (`:sort-order nil`) that prevents meddling with my sort 😂.

File: deps.edn [hunk 3]
```diff
@@ -0,0 +1,8 @@
+;; Shadow mode deps.edn so compliment can be a git or local folder dep
+
+{:paths ["src"]
+ :deps {org.clojure/clojure {:mvn/version "1.11.1"}}
+ 
+ :alises {:dev {:extra-paths ["test"]
+                :extra-deps {fudje/fudje {:mvn/version "0.9.7"}
+                             criterium/criterium {:mvn/version "0.4.6"}}}}}
```
[the remark this answers is not in the document]
Comment (JJ-Atkinson) [reply deps.edn:8]: Nevermind... Unless you know a solution to eastwood looking to `deps.edn` first the CI pipe is broken by this file, so I'll probably just dump it in the interest of expediency.

File: src/compliment/core.clj (hunk 2 above)
Comment (alexander-yakushev) [reply src/compliment/core.clj:72]: I don't mind this, but:

1. Please, at least make it a separate commit.
2. I'd rather see an explicit keyword like `:none`.

File: src/compliment/context.clj (hunk 1 above)
Comment (alexander-yakushev) [reply src/compliment/context.clj:50]: I'd love to see a couple tests for this.

**Comment by JJ-Atkinson** [comment]
Sweet, I think it's all sorted out

**Comment by JJ-Atkinson** [comment]
Thanks for moving to deps.edn btw - it's a great help!

**Comment by JJ-Atkinson** [comment]
@alexander-yakushev anything else that I should look into fixing on this PR? I don't mean to be a nag but I'm hoping to release some code depending on this soon.

**Comment by alexander-yakushev** [comment]
I am sorry, I forgot that this is ready to go. Thanks for the contribution!

**Comment by JJ-Atkinson** [comment]
no problem - thanks!

**Pull request merged**